authV3.go

```diff
@@ -0,0 +1,212 @@
+package circuits
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	core "github.com/iden3/go-iden3-core/v2"
+	"github.com/iden3/go-iden3-crypto/babyjub"
+	"github.com/iden3/go-merkletree-sql/v2"
+	"github.com/pkg/errors"
+)
+
+// AuthV3Inputs type represent authV3.circom/authV3-8-32.circom inputs
+type AuthV3Inputs struct {
+	BaseConfig
+
+	GenesisID    *core.ID `json:"genesisID"`
+	ProfileNonce *big.Int `json:"profileNonce"`
+
+	AuthClaim *core.Claim `json:"authClaim"`
+
+	AuthClaimIncMtp    *merkletree.Proof `json:"authClaimIncMtp"`
+	AuthClaimNonRevMtp *merkletree.Proof `json:"authClaimNonRevMtp"`
+	TreeState          TreeState         `json:"treeState"`
+
+	GISTProof GISTProof `json:"gistProof"`
+
+	Signature *babyjub.Signature `json:"signature"`
+	Challenge *big.Int           `json:"challenge"`
+}
+
+// authV3CircuitInputs type reflect authV3.circom/authV3-8-32.circom private inputs required by prover
+type authV3CircuitInputs struct {
```

Comment:
if it is private and duplicates authV2CircuitInputs we can fully reuse it, or even rename to authCIrcuitInputs